OlinkAnalyze/R/read_npx_parquet.R

```diff
@@ -75,36 +79,32 @@ read_npx_parquet <- function (filename) {
                    "Internal CLI Data Export File",
                    "R Package Export File",
                    "Olink Analyze Export File")
-  if (parquet_file$metadata$DataFileType %in% olink_files) {
-
-    # Check that required columns are present
-    required_cols <- c("SampleID",
-                       "OlinkID",
-                       "UniProt",
-                       "Assay",
-                       "Panel",
-                       "PlateID",
-                       "SampleQC",
-                       "NPX")
-
-    missing_cols <- setdiff(required_cols,
-                            names(parquet_file))
-
-    if(length(missing_cols) != 0) {
-
-      stop(paste("The following columns are missing:",
-                 paste(missing_cols, collapse = ", ")))
-
-    }
-
-  } else {
-
-    stop("Only \"NPX\" parquet files are currently supported.")
+  if (exp_p_meta$dtftp %in% p_file_meta
+      & !(p_file$metadata[[exp_p_meta$dtftp]] %in% olink_files)) {
+    cli::cli_warn("Only {.val {\"NPX\"}} parquet files are currently
+                  supported.")
+  }
 
+  # Check that required columns are present
+  required_cols <- c("SampleID",
+                     "OlinkID",
+                     "UniProt",
+                     "Assay",
+                     "Panel",
+                     "PlateID",
+                     "SampleQC",
+                     "NPX")
+  missing_cols <- setdiff(x = required_cols,
+                          y = names(p_file))
+  if(length(missing_cols) != 0L) {
+    cli::cli_abort(
+      "The following columns are missing: {.val {missing_cols}}"
+    )
   }
```

Comment:
Why do we need to check the column names in the parquet file? Isn't this already done in the read_npx_explore function? This does not break the function in any way, but I believe it is unnecessary.